app/controllers/api/v2/settings_controller.rb

```diff
@@ -53,6 +53,8 @@ def update
         end
         @setting = Foreman.settings.set_user_value(@setting.name, value)
         process_response @setting.save
+      rescue Foreman::SettingValueException => e
+        render_error :custom_error, :locals => { :message => e.message }, :status => :unprocessable_entity
```

Comment:
I put the nice error message in the response here, like so:
```
{
  "error": {"message":"ERF07-1096 [Foreman::SettingValueException]: error parsing value for setting 'entries_per_page': Value is invalid: must be integer"}
}

```
but I had trouble working out how to get anything better on the frontend toast notification than
```
Error updating setting: Request failed with status code 422
```